templates/lxc-alpine.in

```diff
@@ -1,367 +1,505 @@
-release=
-arch=$(uname -m)
+setup_services() {
+	local svc_name
+
+	# Specify the LXC subsystem.
+	sed -i 's/^#*rc_sys=.*/rc_sys="lxc"/' etc/rc.conf
 
-# template mknods, requires root
-if [ $(id -u) -ne 0 ]; then
-   echo "$(basename $0): must be run as root" >&2
-   exit 1
+	# boot runlevel
+	for svc_name in bootmisc hosts syslog; do
+		ln -s /etc/init.d/$svc_name etc/runlevels/boot/$svc_name
+	done
+
+	# default runlevel
+	for svc_name in networking cron; do
+		ln -s /etc/init.d/$svc_name etc/runlevels/default/$svc_name
+	done
+}
+
+
+#===========================  Configure  ===========================#
+
+configure_container() {
+	local config="$1"
+	local hostname="$2"
+	local arch="$3"
+
+	cat <<-EOF >> "$config"
+
+		# Specify container architecture.
+		lxc.arch = $arch
+
+		# Set hostname.
+		lxc.utsname = $hostname
+
+		# If something doesn't work, try to comment this out.
+		# Dropping sys_admin disables container root from doing a lot of things
+		# that could be bad like re-mounting lxc fstab entries rw for example,
+		# but also disables some useful things like being able to nfs mount, and
+		# things that are already namespaced with ns_capable() kernel checks, like
+		# hostname(1).
+		lxc.cap.drop = sys_admin
+
+		# Include common configuration.
+		lxc.include = $LXC_TEMPLATE_CONFIG/alpine.common.conf
+	EOF
+}
+
+
+#=============================  Main  ==============================#
+
+if [ "$(id -u)" != "0" ]; then
+	die 1 "This script must be run as 'root'"
 fi
 
-options=$(getopt -o hn:p:r:R:a: -l help,name:,rootfs:,path:,repository:,release:,arch: -- "$@")
-[ $? -eq 0 ] || usage_err
+# Parse command options.
+options=$(getopt -o a:dFm:n:p:r:h -l arch:,debug,flush-cache,mirror:,name:,\
+path:,release:,rootfs:,help,mapped-uid:,mapped-gid: -- "$@")
 eval set -- "$options"
 
+# Clean variables and set defaults.
+arch="$(uname -m)"
+debug='no'
+flush_cache='no'
+mirror_url=
+name=
+path=
+release=
+rootfs=
+
+# Process command options.
 while [ $# -gt 0 ]; do
-    case "$1" in
-    -h|--help)
-        usage
-        exit 0
-        ;;
-    -n|--name)
-        name=$2
-        ;;
-    --rootfs)
-        rootfs=$2
-        ;;
-    -p|--path)
-        path=$2
-        ;;
-    -r|--repository)
-        repository=$2
-	;;
-    -R|--release)
-        release=$2
-        ;;
-    -a|--arch)
-        arch=$2
-        ;;
-    --)
-	shift
-        break;;
-    esac
-    shift 2
+	case $1 in
+		-a | --arch)
+			arch=$2; shift 2
+		;;
+		-d | --debug)
+			debug='yes'; shift 1
+		;;
+		-F | --flush-cache)
+			flush_cache='yes'; shift 1
+		;;
+		-m | --mirror)
+			mirror_url=$2; shift 2
+		;;
+		-n | --name)
+			name=$2; shift 2
+		;;
+		-p | --path)
+			path=$2; shift 2
+		;;
+		-r | --release)
+			release=$2; shift 2
+		;;
+		--rootfs)
+			rootfs=$2; shift 2
+		;;
+		-h | --help)
+			usage; exit 0
+		;;
+		--)
+			shift; break
+		;;
+		--mapped-[ug]id)
+			die 1 "This template can't be used for unprivileged containers.\n" \
```

Comment:
I’ve removed `\n`, it was leftover from refactoring.

I don’t want to remove `ERROR: `, because it’s a good practice to highlight error messages like this to be better distinguishable from other output of the script.